src/Collection.php

```diff
@@ -452,4 +452,21 @@ public function takeWhile(callable $condition)
     public function map(callable $condition)
     {
     }
+
+    public function reduceRight(callable $callable, $initial = null)
+    {
+        $reverse = array_reverse($this->items);
+        return array_reduce($reverse, $callable, $initial);
+    }
+
+    public function shuffle()
+    {
+        return shuffle($this->items);
+    }
+
+    public function merge(Collection $collection)
+    {
+        $items = $collection->toArray();
+        return $this->addRange($items);
```

Comment:
$items = $this->items;
shuffle($items);

return new static($this->getType(), $items);